Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -444,31 +354,138 @@ namespace AzFramework
+                ti.m_flags |= TF_ONLINE | TF_DEBUGGABLE | TF_RUNNABLE;
+
+                // last target can be loaded in with a persistent id but no network, correct here
+                if (GetDesiredTarget().GetPersistentId() == packet.GetPersistentId())
+                {
+                    m_settings->m_lastTarget.m_networkId = ti.m_networkId;
+                }
+
+                EBUS_QUEUE_EVENT(TargetManagerClient::Bus, TargetJoinedNetwork, ti);
+            }
+
+            if (GetDesiredTarget().GetPersistentId() != packet.GetPersistentId())
+            {
+                SetDesiredTarget(packet.GetPersistentId());
+            }
+        }
+
+        const uint32_t totalBufferSize = packet.GetSize();
+
+        if (m_tmpInboundBufferPos == 0)
+        {
+            m_tmpInboundBuffer.reserve(totalBufferSize);
+        }
+
+        const uint32_t readSize = AZStd::min(totalBufferSize - m_tmpInboundBufferPos, Neighborhood::NeighborBufferSize);
+        memcpy(m_tmpInboundBuffer.begin() + m_tmpInboundBufferPos, packet.GetMessageBuffer().GetBuffer(), readSize);
+        m_tmpInboundBufferPos += readSize;
+        if (m_tmpInboundBufferPos == totalBufferSize)
+        {
+            AZ::IO::MemoryStream msgBuffer(m_tmpInboundBuffer.data(), totalBufferSize, totalBufferSize);
+            TmMsg* msg = nullptr;
+            AZ::ObjectStream::ClassReadyCB readyCB(AZStd::bind(&TargetManagementComponent::OnMsgParsed, this, &msg, AZStd::placeholders::_1,
+                AZStd::placeholders::_2, AZStd::placeholders::_3));
+            AZ::ObjectStream::LoadBlocking(&msgBuffer, *m_serializeContext, readyCB,
+                AZ::ObjectStream::FilterDescriptor(nullptr, AZ::ObjectStream::FILTERFLAG_IGNORE_UNKNOWN_CLASSES));
+            if (msg)
+            {
+                if (msg->GetCustomBlobSize() > 0)
+                {
+                    void* blob = azmalloc(msg->GetCustomBlobSize(), 1, AZ::OSAllocator, "TmMsgBlob");
+                    msgBuffer.Read(msg->GetCustomBlobSize(), blob);
+                    msg->AddCustomBlob(blob, msg->GetCustomBlobSize(), true);
+                }
+                msg->m_senderTargetId = packet.GetPersistentId();
+
+                m_inboxMutex.lock();
+                m_inbox.push_back(msg);
+                m_inboxMutex.unlock();
+                m_tmpInboundBuffer.clear();
+                m_tmpInboundBufferPos = 0;
+            }
+        }
+
+        return true;
+    }
+
+    AzNetworking::ConnectResult TargetManagementComponent::ValidateConnect(
+        [[maybe_unused]] const AzNetworking::IpAddress& remoteAddress,
+        [[maybe_unused]] const AzNetworking::IPacketHeader& packetHeader,
+        [[maybe_unused]] AzNetworking::ISerializer& serializer)
+    {
+        return AzNetworking::ConnectResult::Accepted;
+    }
+
+    void TargetManagementComponent::OnConnect([[maybe_unused]] AzNetworking::IConnection* connection)
+    {
+        ;
```

Comment:
Could we add comment why this is no-op? Ie "no special handshake required, do nothing?" - make it clear its intentionally blank rather than incomplete?